autopilot/agent.go

```diff
@@ -473,17 +475,33 @@ func (a *Agent) controller() {
 		pendingMtx.Unlock()
 
 		// Now that we've updated our internal state, we'll consult our
-		// channel attachment heuristic to determine if we should open
+		// channel attachment heuristics to determine if we should open
 		// up any additional channels or modify existing channels.
-		availableFunds, numChans, needMore := a.cfg.Heuristic.NeedMoreChans(
-			totalChans, a.totalBalance,
-		)
-		if !needMore {
+		var heuristic AttachmentHeuristic
+		var availableFunds btcutil.Amount
+		var numChans uint32
+		for _, h := range a.cfg.Heuristics {
```

Comment:
I would introduce a struct called `CompositeAttachmentHeuristic` that executes this first come first serve logic. The list of child-heuristics is maintained in that struct. Then at least this code doesn't need to be aware of multiple heuristics being active. 

It is then also an option to move the `moreChans` / `distribute` calls in that struct.